src/VecSim/algorithms/brute_force/bf_batch_iterator.cpp

```diff
@@ -134,17 +157,23 @@ VecSimQueryResult_List BF_BatchIterator::getNextResults(size_t n_res,
         for (auto &block : blocks) {
             // compute the scores for the vectors in every block and extend the scores array.
             vecsim_stl::vector<std::pair<float, labelType>> block_scores =
-                block->computeBlockScores(getIndex()->distFunc(), getQueryBlob());
+                this->computeBlockScores(block);
+            if (block_scores.empty()) {
```

Comment:
how about sending an `int*` as an error code to the function and checking it for `VecSim_QueryResult_TimedOut` instead of relying on an empty vector